src/Framework/AspNetCoreAnalyzers/src/CodeFixes/WebApplicationBuilderFixer.cs

```diff
@@ -0,0 +1,165 @@
+                            cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, SyntaxFactory.IdentifierName("Logging")),
+                            equivalenceKey:
+                            DiagnosticDescriptors.DoNotUseHostConfigureLogging.Id),
+                            diagnostic);
+                    break;
+
+                case string when id == DiagnosticDescriptors.DoNotUseHostConfigureServices.Id:
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            "Fix references to Services properties on WebApplicationBuilder",
+                            cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, SyntaxFactory.IdentifierName("Services")),
+                            equivalenceKey:
+                            DiagnosticDescriptors.DoNotUseHostConfigureServices.Id),
+                            diagnostic);
+                    break;
+
+                case string when id == DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder.Id:
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            "Fix references to Configuration properties on WebApplicationBuilder",
+                            cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, SyntaxFactory.IdentifierName("Configuration")),
+                            equivalenceKey:
+                            DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder.Id),
+                            diagnostic);
+                    break;
+            }
+        }
+
+        return Task.CompletedTask;
+    }
+
+    private static async Task<Document> FixWebApplicationBuilder(Diagnostic diagnostic, Document document, CancellationToken cancellationToken, IdentifierNameSyntax identifierMethod)
+    {
+        var editor = await DocumentEditor.CreateAsync(document, cancellationToken);
+        var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+        if (root == null)
+        {
+            return document;
+        }
+
+        //whole invocation starting from "builder"
+        var diagnosticTarget = root.FindNode(diagnostic.Location.SourceSpan); 
+
+        if (diagnosticTarget is InvocationExpressionSyntax originalInvocation)
+        {
+            // if builder.{Host or WebHost}.{method_name} or builder.{Host or WebHost}
+            // is not of type MemberAccessExpressionSyntax, return original code
+            if (originalInvocation.Expression is not MemberAccessExpressionSyntax hostBasedInvocationMethodExpr
+                || hostBasedInvocationMethodExpr.Expression is not MemberAccessExpressionSyntax subExpression)
+            {
+                return editor.OriginalDocument;
```

Comment:
```suggestion
                return document;
```

We can just return the unmodified document here and elsewhere we are returning `editor.OriginalDocument`.